pvlib/pvsystem.py

```diff
@@ -891,29 +891,31 @@ def get_ac(self, model, p_dc, v_dc=None):
         model = model.lower()
         multiple_arrays = self.num_arrays > 1
         if model == 'sandia':
+            p_dc = self._validate_per_array(p_dc)
+            v_dc = self._validate_per_array(v_dc)
             if multiple_arrays:
-                p_dc = self._validate_per_array(p_dc)
```

Comment:
why remove the `_validate_per_array` here?

Reply:
If `p_dc` and `v_dc` are floats they will get wrapped up as `(p_dc,)` and `(v_dc,)`. Doing this regardless of the number of arrays lets us always subscript when passing to the single array inverter models. If we didn't wrap them up we would need extra logic to handle unwrapping if the input was initially a singleton tuple.

The goal is to be able to do both of these and have them act the same:

```python
system = PVSystem(arrays=[Array()], ...)
system.get_ac(p_dc=50, model='pvwatts')
system.get_ac(p_dc=(50,), model='pvwatts')
```

Reply:
As written, `get_ac` references the `inverter` functions directly, not through a `PVSystem` method for each inverter function. The `inverter` functions don't know what to do with a tuple of values; they assume `numeric`.

Reply:
The tuples are *always* subscripted before being passed to the inverter functions when `multiple_arrays` is `False`. For example:
```python
        if model == 'sandia':
            p_dc = self._validate_per_array(p_dc)
            v_dc = self._validate_per_array(v_dc)
            if multiple_arrays:
                return inverter.sandia_multi(
                    v_dc, p_dc, self.inverter_parameters)
            return inverter.sandia(v_dc[0], p_dc[0], self.inverter_parameters)
```
When `multiple_arrays` is True the tuples are passed directly to `inverter.sandia_multi`, but `p_dc[0]` and `v_dc[0]` is used for `inverter.sandia`.

Reply:
I think we can now move the `_validate_per_array` calls above the `if model ==...` block and save a few lines:

```python
p_dc = self._validate_per_array(p_dc)
if v_dc is not None:
    v_dc = self._validate_per_array(v_dc)
```